python/ray/data/preprocessors/hasher.py

```diff
@@ -75,11 +75,22 @@ class FeatureHasher(Preprocessor):
 
     _is_fittable = False
 
-    def __init__(self, columns: List[str], num_features: int):
+    def __init__(
+        self,
+        columns: List[str],
+        num_features: int,
+        output_columns: Optional[List[str]] = None,
```

Comment:
@richardliaw wondering what your thoughts are on these changes. The `FeatureHasher` already create new columns, but it drops the original columns from the result dataset.
How can we enable/disable the dropping of the columns?

In these changes I opted for having the `output_columns`, which should name the transformed columns (instead of naming them `hash_{i}`) and if the output columns are provided, it will also make it so that the hasher doesn't drop the original columns.

Not super happy with this, there are a few other options we could take, like having an `append`, that controls whether to drop the original or not, but that won't allow us to name the output columns differently (as we do on other preprocessors)

Reply:
Hmm, I think it makes sense to only have a single `output_column` here, and disable the dropping of columns altogether?

Reply:
@richardliaw not sure I follow. Right now the hasher preprocessor outputs num_features new columns. Are you saying that you want to change it so it always returns a single column with a a list of `num_features` elements?

Reply:
yes, this would save us from headache in situations with wider tables

Reply:
done in https://github.com/ray-project/ray/pull/50632/commits/ec6f311e33c3dc34f1ec58794edd6267984b5521